lib/rspec/core/configuration.rb

```diff
@@ -870,7 +874,8 @@ def full_description
       # Adds a formatter to the set RSpec will use for this run.
       #
       # @see RSpec::Core::Formatters::Protocol
-      def add_formatter(formatter, output=output_stream)
+      def add_formatter(formatter, output=output_wrapper)
+        output = OutputWrapper.new(output) unless output == output_wrapper
```

Comment:
Now the code of `Configuration#add_formatter` is:

```ruby
def add_formatter(formatter, output=output_stream)
  formatter_loader.add(formatter, output)
end
```

As you may see, `add_formatter` does change the `output_stream` in any way. If we set the `output` param to the existing wrapper, it is equivalent to the following code (in the version without wrapper):

```ruby
def add_formatter(formatter, output=output_stream)
  @output_stream = output
  formatter_loader.add(formatter, @output_stream)
end
```

It changes the current behaviour of `add_formatter` where you could couple any formatter with a specific `output`, which does not relate to the common `Configuration#output_stream`. In my opinion that behavior was quite logical.

Also i did not find any specific documentation or a test about this case. And if you change `add_formatter` to:

```ruby
def add_formatter(formatter, output=output_stream)
  @output_stream = output
  formatter_loader.add(formatter, @output_stream)
end
```

all specs are still green meaning it’s not covered by tests. 
But i’m not sure it is reasonable to change the current behaviour of `add_formatter` here.